git-branchless-navigation/src/lib.rs

```diff
@@ -512,16 +512,22 @@ pub fn switch(
 
     enum Target {
         Interactive(String),
+        Revision(String),
```

Comment:
issue: I think it's misleading to call this `Revision`, because it sounds like it's meant to refer to a revision/commit specifically, and not be a general revision specification, which can indicate a branch.

Some ideas for renaming it:
- `Revspec` (but I'm not sure if this is legitimate Git terminology)
- `RevisionOrRef`
- `RevparseSingleCommit`
- `Uninterpreted` or similar
  - I guess it would be better if `CheckoutTarget::Unknown` were actually named `CheckoutTarget::Uninterpreted` or similar, since `Unknown` is kind of weird to say when the value is clearly known.

Let's also add a comment to this `enum` case explaining that it can also represent branches, etc.

Reply:
Good call. I ended up with `Passthrough` here and documented as follows:
```
        /// The target expression is probably a git revision or reference and
        /// should be passed directly to git for resolution.
        Passthrough(String),
```

I was tempted by `RevisionOrRef` but wasn't sure if that might also fall into the same trap as just `Revision`. I too am not sure on my git terminology on this one, though, so maybe  revision and ref really are the only two options?

Happy to change this again if you'd like.